dpytools/config/properties/base.py

```diff
@@ -9,13 +9,24 @@ class BaseProperty(metaclass=ABCMeta):
 
     # TODO: getter
     # allow someone to get the property
+    def _get_name(self):
```

Comment:
so there's another python trick here in regards to using the `@property` decorator. If you rewrite these get and set methods using the following patterns:

```
    @property
    def name(self):
        return self._name

    @name.setter
    def name(self, value):
        raise ValueError(f"Trying to change name property to value {value} but you cannot change a property name after instantiation.")
```

it results in the behaviour:

- `print(config.name1.name)` will work, the first method will be called and (under the hood) the value of self._name will be returned.
- wheres `config.name1.name = "some new value"` would result in the ValueError being raised.

So just change the base class `name` and `value` fields to be `_name` and `_value` and you can use the property decorator to protect us against anyone trying to change a config value after the application has started.

Defensive programming pretty much. It's usually overkill in python but someone changing config values after an app has started would be a very hard issue to debug so we're gonna get ahead of it.

As part of this could you add a test to confirm that nobody can change a property value after the property class has been instantiated please.

In hindsight I do realise by initial sketch of base class did not help to convey this idea 😄

Reply:
Can we remove the TODO's as well please.